mubench.reviewsite/src/Controllers/ReviewsController.php

```diff
@@ -131,7 +139,55 @@ public function updateOrCreateReview($misuse_id, $reviewer_id, $comment, $findin
             $findingReview = FindingReview::firstOrNew(['review_id' => $review->id, 'rank' => $rank]);
             $findingReview->decision = $findings_review['hit'];
             $findingReview->save();
-            $findingReview->violation_types()->sync($findings_review['types']);
+            if($findings_review['types']){
+                $findingReview->violation_types()->sync($findings_review['types']);
+            }
+        }
+    }
+
+    private function collectAllMisuses($runs)
+    {
+        $all_misuses = new Collection;
+        foreach ($runs as $run) {
+            $all_misuses = $all_misuses->merge($run->misuses);
+        }
+        return $all_misuses;
+    }
+
+    private function determineNavigationTargets(Collection $all_misuses, $project_muid, $version_muid, $misuse_muid, $reviewer)
+    {
+        $previous_misuse = $all_misuses->last();
+        $misuse = NULL;
+        $next_misuse = NULL;
+        $next_reviewable_misuse = NULL;
+
+        foreach ($all_misuses as $current_misuse) { /** @var Misuse $current_misuse */
+            if (!$misuse && $current_misuse->getProject() === $project_muid
+                && $current_misuse->getVersion() === $version_muid
+                && $current_misuse->misuse_muid === $misuse_muid) {
+                $misuse = $current_misuse;
+            } else {
+                if ($misuse && !$next_misuse) {
+                    $next_misuse = $current_misuse;
+                }
+                if ((!$next_reviewable_misuse || $misuse)
```

Comment:
I'm not sure why this condition is how it is. Please give this a speaking name, either through an intermediate variable or a helper method. Btw. why are there no tests for this logic?

Reply:
next_reviewable_misuse is null before going through all the misuses. We set it on the first available misuse that is reviewable or on the first reviewable misuse after we already found the current misuse.

Reply:
Oh, I get it. So this means `if (isFirstReviewableMisuse($current_misuse) || isFirstReviewableMisuseAfter($current_misuse, $misuse))`. Make sense, but did not reveal itself to me. Please use some intermediate variables or helper functions to give names to these intentions.